lib/ecto/query/builder/cte.ex

```diff
@@ -95,10 +95,23 @@ defmodule Ecto.Query.Builder.CTE do
 
   @doc false
   def apply_cte(nil, name, with_query, materialized) do
-    %Ecto.Query.WithExpr{queries: [{name, materialized, with_query}]}
+    %Ecto.Query.WithExpr{queries: [{%{name: name, materialized: materialized}, with_query}]}
```

Comment:
Ah, I did not know the name is used for merging. My suggestion then is to keep it as:

```
{name, map, with_query}
```

WDYT?

And then we only include materialize in the map if it is a boolean?